modules/message-lane/src/lib.rs

```diff
@@ -348,6 +352,59 @@ decl_module! {
 			Ok(())
 		}
 
+		/// Pay additional fee for the message.
+		#[weight = T::WeightInfo::increase_message_fee()]
+		pub fn increase_message_fee(
+			origin,
+			lane_id: LaneId,
+			nonce: MessageNonce,
+			additional_fee: T::OutboundMessageFee,
+		) -> DispatchResult {
+			// if someone tries to pay for already-delivered message, we're rejecting this intention
+			// (otherwise this additional fee will be locked forever in relayers fund)
+			//
+			// if someone tries to pay for not-yet-sent message, we're rejeting this intention, or
+			// we're risking to have mess in the storage
+			let lane = outbound_lane::<T, I>(lane_id);
+			ensure!(
+				nonce > lane.data().latest_received_nonce && nonce <= lane.data().latest_generated_nonce,
+				Error::<T, I>::InactiveMessage
+			);
```

Comment:
I'd be tempted to split this into two different errors, just to ensure that the caller knows precisely what went wrong